eclipse-theia/theia-cpp-extensions
# cpp: fix build issues with newer theia

#### What it does


The commit fixes build issues when using newer versions of `@theia` extensions:
- updates the repo to the latest theia (`v1.17.2`).
- updates target from `es5` to `es2017`.

Previous versions of the extension were causing runtime issues on docker (theia-cpp-docker).

#### How to test


- confirm the application builds
- confirm the application starts
- confirm that there are no previous `es5` issues, that the top-level menu successfully displays

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
